test/sharness/lib/test-lib.sh

```diff
@@ -45,7 +45,9 @@ test "$TEST_EXPENSIVE" = 1 && test_set_prereq EXPENSIVE
 test "$TEST_NO_DOCKER" != 1 && type docker && test_set_prereq DOCKER
 
 TEST_OS=$(uname -s | tr [a-z] [A-Z])
-test expr "$TEST_OS" : "CYGWIN_NT" >/dev/null && test_set_prereq POSIX
+
+# Set a prereq as error messages are often different on Windows/Cygwin
+expr "$TEST_OS" : "CYGWIN_NT" >/dev/null || test_set_prereq STD_ERR_MSG
```

Comment:
I agree: error messages are highly unstandardized and not very portable. If we can get away with just checking error codes on the more uncommon platforms we should do it. My concern was with choosing a name that communicated its intent well: having globals is bad enough, so choosing really specific, easily understood names is important.

> Previously I used "POSIX", so perhaps "POSIX_ERR_MSG" is better, otherwise I am open to suggestions.

Why not just keep the `POSIX` prereq and test for that? Is there value in differentiating between "is posix" and "has posix error messages"?